src/components/CodeSnippet/CodeSnippet.js

```diff
@@ -1,43 +1,144 @@
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
 import classNames from 'classnames';
+import Copy from '../Copy';
 import CopyButton from '../CopyButton';
+import Icon from '../Icon';
 
-const CodeSnippet = ({
-  className,
-  type,
-  children,
-  feedback,
-  onClick,
-  wrappedContentRef,
-  ...other
-}) => {
-  const snippetType =
-    type === 'terminal' ? 'bx--snippet--terminal' : 'bx--snippet--code';
-  const wrapperClasses = classNames('bx--snippet', className, snippetType);
-  return (
-    <div className={wrapperClasses} {...other}>
+export default class CodeSnippet extends Component {
+  static propTypes = {
+    /**
+     * The type of code snippet
+     * can be inline, single or multi
+     */
+    type: PropTypes.string,
+    className: PropTypes.string,
+    children: PropTypes.string,
+    feedback: PropTypes.string,
+    copyLabel: PropTypes.string,
+    onClick: PropTypes.func,
+    wrappedContentRef: PropTypes.func,
+    showMoreText: PropTypes.string,
+    showLessText: PropTypes.string,
+    /**
+     * Used with inline snippet only
+     * to display alternate color
+     */
+    light: PropTypes.bool,
+  };
+
+  static defaultProps = {
+    type: 'single',
+    showMoreText: 'Show more',
+    showLessText: 'Show less',
+  };
+
+  state = {
+    showBtn: false,
+    expandedCode: false,
+  };
+
+  componentDidMount = () => {
+    if (ReactDOM.findDOMNode(this).offsetHeight > 190) {
+      this.setState({ showBtn: true });
+    }
+  };
+
+  expandCode = () => {
+    this.setState({ expandedCode: !this.state.expandedCode });
+  };
+
+  render() {
+    const {
+      className,
+      type,
+      children,
+      feedback,
+      onClick,
+      copyLabel,
+      wrappedContentRef,
+      light,
+      /*
+      showMoreText,
+      showLessText,
+      */
+      ...other
+    } = this.props;
+
+    const codeSnippetClasses = classNames(className, {
+      'bx--snippet': true,
+      'bx--snippet--single': type === 'single' || type === 'terminal',
+      'bx--snippet--multi': type === 'multi' || type === 'code',
+      'bx--snippet--inline': type === 'inline',
+      'bx--snippet--expand': this.state.expandedCode,
+      'bx--snippet--light': light,
+    });
+
+    const expandCodeBtnText = this.state.expandedCode ? 'less' : 'more'; // need the show/more less text props to show here
```

Comment:
@joshblack I was struggling with the best way to show these text values, it wouldn't let me use the props. Any ideas?